packages/internal/test-util/src/lib/noop-driver/noop-driver-root.module.ts

```diff
@@ -1,38 +1,22 @@
-import { NgModule, Optional, SkipSelf } from '@angular/core';
+import { inject, InjectFlags, NgModule } from '@angular/core';
 
-import {
-  LumberjackLogDriverConfig,
-  lumberjackLogDriverConfigToken,
-  lumberjackLogDriverToken,
-} from '@ngworker/lumberjack';
+import { lumberjackLogDriverToken } from '@ngworker/lumberjack';
 
-import { noopDriverConfigToken } from './noop-driver-config.token';
-import { NoopDriverConfig } from './noop-driver.config';
 import { NoopDriver } from './noop.driver';
 
-export function noopDriverFactory(
-  logDriverConfig: LumberjackLogDriverConfig,
-  noopDriverConfig: NoopDriverConfig
-): NoopDriver {
-  const baseConfig = { ...logDriverConfig, identifier: NoopDriver.driverIdentifier };
-  const fullConfig = { ...baseConfig, ...noopDriverConfig };
-
-  return new NoopDriver(fullConfig);
-}
-
 @NgModule({
   providers: [
     {
       provide: lumberjackLogDriverToken,
-      useFactory: noopDriverFactory,
-      deps: [lumberjackLogDriverConfigToken, noopDriverConfigToken],
+      useClass: NoopDriver,
       multi: true,
     },
   ],
 })
 export class NoopDriverRootModule {
-  constructor(@Optional() @SkipSelf() maybeNgModuleFromParentInjector?: NoopDriverRootModule) {
-    if (maybeNgModuleFromParentInjector) {
+  maybeNgModuleFromParentInjector? = inject(NoopDriverRootModule, InjectFlags.SkipSelf | InjectFlags.Optional);
```

Comment:
nit: separate property and constructor with a blank line.

Reply:
issue: mark as private.
```suggestion
  private maybeNgModuleFromParentInjector? = inject(NoopDriverRootModule, InjectFlags.SkipSelf | InjectFlags.Optional);
```

Reply:
issue: we can't use the optional operator (`?`) because optional dependencies have a default value of `null`, not `undefined`.
```suggestion
  maybeNgModuleFromParentInjector = inject(NoopDriverRootModule, InjectFlags.SkipSelf | InjectFlags.Optional);
```